runtime-modules/content/src/lib.rs

```diff
@@ -708,22 +720,29 @@ decl_module! {
                 )?;
             }
 
-            //
-            // == MUTATION SAFE ==
-            //
-
             let mut channel = channel;
 
-            // Maybe update the reward account
-            if let Some(reward_account) = &params.reward_account {
+            // maybe update the reward account if actor is not a collaborator
+            if let Some(reward_account) = params.reward_account.as_ref() {
+                ensure_actor_not_a_collaborator::<T>(&actor)?;
```

Comment:
This check should be done before calls to `Self::remove_assets_from_storage` and `Self::upload_assets_to_storage`.

When I tried to update both channel avatar and `reward_account` as `Collaborator` I noticed that:
- the old avatar is removed
- the new avatar is uploaded, but it's not assigned to the channel by the query node, because `ChannelUpdated` event is never emitted
- extrinsic fails with `ActorNotAuthorized`, but the runtime state is changed
- this results in channel having no assigned avatar and a dangling data object of `DataObjectTypeUnknown` can be seen in query node

Events emitted as a result of the call:
```
balances.Transfer
storage.BagObjectsChanged
storage.VoucherChanged
storage.DataObjectsDeleted
balances.Transfer
storage.BagObjectsChanged
storage.VoucherChanged
storage.DataObjectsUploaded
system.ExtrinsicFailed
```

Please make sure no state changes are done before all of extrinsic parameters are validated. (From what I understand that's why all methods have the `// == MUTATION SAFE ==` comment)